src/toil/batchSystems/lsf.py

```diff
@@ -167,18 +169,18 @@ def prepareBsub(cpu, mem, jobID):
                     mem = old_div(float(mem), 1024**3)
                     mem_resource = parse_memory_resource(mem)
                     mem_limit = parse_memory_limit(mem)
-                bsubMem = ['-R', 'select[type==X86_64 && mem > {m}] '
+                bsubMem = ['-R',
                            'rusage[mem={m}]'.format(m=mem_resource),
                            '-M', str(mem_limit)]
             else:
                 bsubMem = []
             bsubCpu = [] if cpu is None else ['-n', str(math.ceil(cpu))]
-            bsubline = ["bsub", "-cwd", ".", "-J", "toil_job_{}".format(jobID)]
+            bsubline = ["bsub", "-cwd", ".",
+                        "-J", "toil_job_{}".format(jobID),
+                        "-o", "toil_job_{}.out".format(jobID),
+                        "-e", "toil_job_{}.err".format(jobID)]
             bsubline.extend(bsubMem)
             bsubline.extend(bsubCpu)
-            stdoutfile = self.boss.formatStdOutErrPath(jobID, 'lsf', '%J', 'std_output')
```

Comment:
Why are these function calls dropped?

Reply:
Because the stdout and stderr were added to the bsub command line call above

Reply:
I prefer to keep the `formatStdOutErrPath` call. It will add `-e` and `-o` flags like this does but it uses the abstract method which would be consistent with other schedulers. Unless there is a reason this method doesn't work correctly for LSF I'm in favor of keeping it as it was.

Reply:
I can add it back, I just didn't understand why I needed it if I had the -o and -e included in the line above. Should I remove it from the line:

``` 
          bsubline = ["bsub", "-cwd", ".", 
                        "-J", "toil_job_{}".format(jobID), 
                        "-o", "toil_job_{}.out".format(jobID), 
                        "-e", "toil_job_{}.err".format(jobID)]
```
I guess I am not understanding what the `self.boss.formatStdOutErrPath` is doing

Reply:
@drkennetz It names the output files in a consistent manner across batch schedulers and puts them in the work directory instead of the current working directory as your code does

https://github.com/DataBiosphere/toil/blob/ff31650c0eb0b5438b89002f789ce46b85cccc6a/src/toil/batchSystems/abstractBatchSystem.py#L304

https://github.com/DataBiosphere/toil/blob/ff31650c0eb0b5438b89002f789ce46b85cccc6a/src/toil/common.py#L1009:9

Reply:
my code does indeed. I'll add the two recommendations above back into the code, run them again and see if outputs are affected. If not I'll commit changes back.

Thanks for your help! @drjrm3 @mr-c

Reply:
Sorry for not documenting this as I was going through the first time. This is why I changed the code here:

```
            stdoutfile = self.boss.formatStdOutErrPath(jobID, 'lsf', '%J', 'std_output')
            stdoutfile = self.boss.formatStdOutErrPath(jobID, 'lsf', '%J', 'std_output')
            bsubline.extend(['-o', stdoutfile, '-e', stderrfile])
```
`self` is not defined:

```
Traceback (most recent call last):
  File "/hpcf/apps/python/install/3.5.2/lib/python3.5/threading.py", line 914, in _bootstrap_inner
    self.run()
  File "/users/dkennetz/mytoil/lib/python3.5/site-packages/toil/batchSystems/abstractGridEngineBatchSystem.py", line 234, in run
    activity |= self.createJobs(newJob)
  File "/users/dkennetz/mytoil/lib/python3.5/site-packages/toil/batchSystems/abstractGridEngineBatchSystem.py", line 136, in createJobs
    subLine = self.prepareSubmission(cpu, memory, jobID, command)
  File "/users/dkennetz/mytoil/lib/python3.5/site-packages/toil/batchSystems/lsf.py", line 71, in prepareSubmission
    return self.prepareBsub(cpu, memory, jobID) + [command]
  File "/users/dkennetz/mytoil/lib/python3.5/site-packages/toil/batchSystems/lsf.py", line 182, in prepareBsub
    stdoutfile = self.boss.formatStdOutErrPath(jobID, 'lsf', '%J', 'std_output')
NameError: name 'self' is not defined
```

Reply:
I have fixed the indent on the other comment, but after changing the `bsubline` to my fix, I get a successful workflow.

Reply:
@drkennetz I think `self` is not defined because this is defined as a `@staticmethod`. I'm not sure what benefit the static nature is providing but I think we will need to remove that if we want to use `self.boss.formatStdOutErrPath`. @mr-c - thoughts?

Reply:
Yeah this makes sense. If I remove `@staticmethod` and add `self` into the function arguments I get success.

Reply:
The log files look like:
```
[dkennetz@hpc02  toil-f654dd2a-ae66-492f-a246-3a15c1dee365-cc963444-5205-4a81-8502-3445f286bed5]$ ls
toil_job_1_batch_lsf_81672902_std_error.log   toil_job_2_batch_lsf_81672903_std_error.log   toil_job_3_batch_lsf_81672904_std_error.log
toil_job_1_batch_lsf_81672902_std_output.log  toil_job_2_batch_lsf_81672903_std_output.log  toil_job_3_batch_lsf
```
and persist as long as whatever I have the `--cleanWorkDir` flag set to.